File: app/core/src/main/java/com/fsck/k9/helper/EmailHelper.java
package com.fsck.k9.helper;


public final class EmailHelper {
    private EmailHelper() {}

    public static String getDomainFromEmailAddress(String email) {
        int separatorIndex = email.lastIndexOf('@');
        if (separatorIndex == -1 || separatorIndex + 1 == email.length()) {
            return null;
        }

        return email.substring(separatorIndex + 1);
    }
}

File: app/core/src/main/java/com/fsck/k9/helper/EmailHelper.kt
package com.fsck.k9.helper

object EmailHelper {
    @JvmStatic
    fun getDomainFromEmailAddress(email: String): String? {
        val index = email.lastIndexOf('@')
        return if (index == -1 || index == email.lastIndex) null else email.substring(index + 1)
    }
}
